lib/src/arithmetic/values/fixed_point_value.dart

```diff
@@ -0,0 +1,167 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// fixed_point_value.dart
+//
+// 2024 September 24
+// Authors:
+//  Soner Yaldiz <soner.yaldiz@intel.com>
+
+import 'dart:math';
+import 'package:meta/meta.dart';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// An immutable representation of (un)signed fixed-point values following
+/// Q notation (Qm.n format) as introduced by
+/// (Texas Instruments)[https://www.ti.com/lit/ug/spru565b/spru565b.pdf].
+@immutable
+class FixedPointValue implements Comparable<FixedPointValue> {
+  /// The full fixed point value bit storage in two's complement.
+  late final LogicValue value;
+
+  /// The sign of the fixed point number.
+  final LogicValue sign;
+
+  /// The integer part of the fixed-point number.
+  final LogicValue integer;
+
+  /// The fractional part of the fixed-point number.
+  final LogicValue fraction;
+
+  /// Constructs [FixedPointValue] from sign, integer and fraction values.
+  FixedPointValue(
+      {this.sign = LogicValue.empty,
+      this.integer = LogicValue.empty,
+      this.fraction = LogicValue.empty}) {
+    if ((integer == LogicValue.empty) & (fraction == LogicValue.empty)) {
+      throw RohdHclException('integer or fraction must be non-empty');
+    }
+    if ((sign == LogicValue.empty) | (sign.isZero)) {
+      value = [sign, integer, fraction].swizzle();
+    } else {
+      value = ~[LogicValue.zero, integer, fraction].swizzle() + 1;
```

Comment:
doesn't this leave the `integer` and `fraction` with incorrect values that don't match `value`?

Reply:
`value` is always in two's complement where `integer` and `fraction` are not. Accessing `fractional` or `integer` in two's complement fashion is pointless. Assuming that `FloatingPointValue` objects will mostly be constructed using `ofDouble`, this class can be refactored as follows:
```dart
class FixedPointValue implements Comparable<FixedPointValue> {
  /// The full fixed point value bit storage in two's complement.
  late final LogicValue value;
  /// [signed] indicates whether the representation is signed.
  final bool signed;
  /// [m] is the number of bits reserved for the integer part.
  final int m;
  /// [n] is the number of bits reserved for the fractional part.
  final int n;
```
where value is always in two's complement. What do you think?

Reply:
is it not also useful to provide an accessor for the integer and fraction portion of the `value`?  i.e. yes, this looks good, but maybe also add something that `getRange`s the integer and fraction portion?